analyzers.py

```diff
@@ -130,45 +129,52 @@ class CobsXDecoder(HighLevelAnalyzer):
             "format": "ERROR: {{data.error}}"
         }
     }
+    number_of_prefix_bytes_after_0_byte = NumberSetting(min_value = 0, max_value = 10)
 
     def decode_bytes(self, x):
         pass
 
     def __init__(self):
         self.received = []
         self.frame_start_time = None
+        self.frame_end_time = None
 
     def decode(self, frame: AnalyzerFrame):
+        retval = None
         data = frame.data["data"]
 
         if self.frame_start_time is None:
             self.frame_start_time = frame.start_time
 
         if data == b"\0":
-            received, self.received = self.received, []
-            frame_start_time, self.frame_start_time = self.frame_start_time, None
-            try:
-                data = self.decode_bytes(b"".join(received))
-                return AnalyzerFrame(
-                    "Message",
-                    frame_start_time,
-                    frame.end_time,
-                    {
-                        "data": data,
-                    }
-                )
-            except DecodeError as de:
-                return AnalyzerFrame(
-                    "Error",
-                    frame_start_time,
-                    frame.end_time,
-                    {
-                        "error": str(de),
-                    }
-                )
-            
+            if len(self.received) >= self.number_of_prefix_bytes_after_0_byte:
+                del self.received[:int(self.number_of_prefix_bytes_after_0_byte)]
+            if self.received:
+                try:
+                    data = self.decode_bytes(b"".join(self.received))
+                    retval = AnalyzerFrame(
+                        "Message",
+                        self.frame_start_time,
+                        self.frame_end_time,
+                        {
+                            "data": data,
+                        }
+                    )
+                except DecodeError as de:
+                    retval = AnalyzerFrame(
+                        "Error",
+                        self.frame_start_time,
+                        self.frame_end_time,
+                        {
+                            "error": str(de),
+                        }
+                    )
+            self.received = []
+            self.frame_start_time = frame.start_time
```

Comment:
Should set `self.frame_end_time = frame.end_time` here too, in case another null byte is sent (i.e., empty frame)

Reply:
True, fixed it.